Sources/PT.PM.Common/PathUtils.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using System.IO;
+
+namespace PT.PM.Common.Utils
+{
+    public static class PathUtils
+    {
+        public const int MaxDirLength = 248 - 1;
+        public const int MaxPathLength = 260 - 1;
+
+        public static string NormalizeFilePath(this string path) => NormalizePath(path, false);
+
+        public static string NormalizeDirPath(this string path, bool force = false) => NormalizePath(path, true, force);
+
+        private static string NormalizePath(this string path, bool isDirectory = true, bool force = false)
+        {
+            if (CommonUtils.IsWindows && !CommonUtils.IsCoreApp && !path.StartsWith(@"\\?\") &&
+                (path.Length > (isDirectory ? MaxDirLength : MaxPathLength) || force))
+            {
+                if (path.StartsWith(@"\\"))
+                {
+                    return $@"\\?\UNC\{path.Remove(2)}";
+                }
+
+                path = path.NormalizeDirSeparator();
+
+                return $@"\\?\{path}";
+            }
+
+            return path.NormalizeDirSeparator();
+        }
+
+        public static string NormalizeDirSeparator(this string path)
+        {
+            string notPlatformSeparator = CommonUtils.IsWindows ? "/" : "\\";
+
+            if (path.Contains(notPlatformSeparator))
```

Comment:
you shouldn't check for separatorChar contains in path, `Replace` method will make it for you

Reply:
I hope this is an optimization, but not sure. It's rather rare case when slashes removed, so I decided to insert such a check. But maybe you are right and it's useless.